src/metabase/query_processor/streaming/json.clj

```diff
@@ -75,43 +78,62 @@
   ([_]   (qp.si/stream-options :api nil))
   ([_ _] {:content-type "application/json; charset=utf-8"}))
 
-(defn- map->serialized-json-kvs
-  "{:a 100, :b 200} ; -> \"a\":100,\"b\":200"
-  ^String [m]
-  (when (seq m)
-    (let [s (json/generate-string m)]
-      (.substring s 1 (dec (count s))))))
+(defn- generate-map-contents [^JsonGenerator jgen maplike]
+  (reduce (fn [^JsonGenerator jg kv]
+            (let [k (key kv)
+                  v (val kv)]
+              (.writeFieldName jg (if (keyword? k)
+                                    (subs (str k) 1)
+                                    (str k)))
```

Comment:
nit:  Could `if` be reduced to `(name k)`?

Reply:
No, because we want `"foo/bar"`, not `"bar"`.